proving_system/stark/src/cairo_vm/instruction_offsets.rs

```diff
@@ -1,54 +1,66 @@
-use super::cairo_mem::CairoMemoryCell;
+use crate::FE;
+use lambdaworks_math::field::{element::FieldElement, traits::IsField};
+
+use super::instruction_flags::aux_get_last_nim_of_field_element;
 
 const OFF_DST_OFF: u32 = 0;
 const OFF_OP0_OFF: u32 = 16;
 const OFF_OP1_OFF: u32 = 32;
 const OFFX_MASK: u64 = 0xFFFF;
 
+#[derive(Debug, PartialEq, Eq)]
 pub struct InstructionOffsets {
     pub off_dst: i32,
     pub off_op0: i32,
     pub off_op1: i32,
 }
 
 impl InstructionOffsets {
-    pub fn new(cell: &CairoMemoryCell) -> Self {
+    pub fn new(mem_value: &FE) -> Self {
         Self {
-            off_dst: Self::decode_offset(cell, OFF_DST_OFF),
-            off_op0: Self::decode_offset(cell, OFF_OP0_OFF),
-            off_op1: Self::decode_offset(cell, OFF_OP1_OFF),
+            off_dst: Self::decode_offset(mem_value, OFF_DST_OFF),
+            off_op0: Self::decode_offset(mem_value, OFF_OP0_OFF),
+            off_op1: Self::decode_offset(mem_value, OFF_OP1_OFF),
         }
     }
 
-    fn decode_offset(cell: &CairoMemoryCell, instruction_offset: u32) -> i32 {
-        let offset = cell.value.limbs[3] >> instruction_offset & OFFX_MASK;
+    pub fn decode_offset(mem_value: &FE, instruction_offset: u32) -> i32 {
+        let offset = aux_get_last_nim_of_field_element(mem_value) >> instruction_offset & OFFX_MASK;
         let vectorized_offset = offset.to_le_bytes();
         let aux = [
             vectorized_offset[0],
             vectorized_offset[1].overflowing_sub(128).0,
         ];
         i32::from(i16::from_le_bytes(aux))
     }
+
+    pub fn to_trace_representation<F: IsField>(&self) -> [FieldElement<F>; 3] {
+        [
+            to_unbiased_representation(self.off_dst),
+            to_unbiased_representation(self.off_op0),
+            to_unbiased_representation(self.off_op1),
+        ]
+    }
+}
+
+fn to_unbiased_representation<F: IsField>(n: i32) -> FieldElement<F> {
```

Comment:
It's not clear what an unbiased representation is, we could add documentation to this function, maybe the same for trace representation